### PR TITLE
Add 'Speak Message' accessibility feature

### PR DESCRIPTION
**Pull request by meherkasam**
### Contributor checklist
- [x] My commits are rebased on the latest main branch
- [x] My commits are in nice logical chunks
- [ ] My contribution is fully baked and is ready to be merged as is
- [x] I have tested my contribution on these devices:
 * iPhone 11 Pro, iOS 15.6
 * iPhone 13 Pro Simulator, iOS 15.5

- - - - - - - - - -

### Description

This feature was requested on the Signal Community forum [here](https://community.signalusers.org/t/option-to-speak-messages/27113/).

This PR adds support for the Spoken Content accessibility feature (Settings > Accessibility > Spoken Content, toggle on Speak Selection). Once this feature is enabled, message bubbles on iMessage (WhatsApp too) support speaking out the text inside a selected bubble. This is useful for many people including those with visual impairments but don't use VoiceOver and those with dyslexia.

I attempted to replicate the user experience on iMessage with the "Stop" feature (on iMessage, it's called "Pause" but it behaves more like a stop button, hence the change in name).

| iMessage (existing)  | Signal (new) |
| ------------- | ------------- |
| <video src="https://user-images.githubusercontent.com/450030/182534529-fa66db86-d241-452f-81ed-3014e0c5d672.mp4">  | <video src="https://user-images.githubusercontent.com/450030/182534566-de3a8b0f-ac5b-4b57-a2c2-edbcef1951c4.mp4">  |

### CODE REVIEW

File: SignalUI/Appearance/Theme+OWS.swift [hunk 1]
```diff
@@ -293,6 +295,11 @@ public extension Theme {
             return isDarkThemeEnabled ? "camera-solid-24" : "camera-outline-24"
         case .micButton:
             return isDarkThemeEnabled ? "mic-solid-24" : "mic-outline-24"
+        case .speakerButton:
+            return "speaker-solid-28" // TODO: Add outline for dark theme or select another asset.
```
Comment (meherkasam) [inline SignalUI/Appearance/Theme+OWS.swift:299]: - [ ] Fix assets

**Comment by meherkasam** [comment]
@EvanHahn-Signal Do you have any questions or follow-up requests for me on this PR?

**Comment by EvanHahn-Signal** [comment]
Nothing for you right now. We want to make sure this doesn't send messages to Apple, which we're looking into.

**Comment by EvanHahn-Signal** [comment]
Taking another look at this now.

**Comment by EvanHahn-Signal** [comment]
We'll merge this into our internal repository and deploy this in an upcoming version. Thanks for doing this!

File: SignalUI/Appearance/Theme+OWS.swift [hunk 2]
```diff
@@ -293,6 +295,11 @@ public extension Theme {
...
+        case .stopButton:
+            return "pause-filled-24" // TODO: Use correct asset
```
Comment (meherkasam) [inline SignalUI/Appearance/Theme+OWS.swift:301]: @EvanHahn-Signal I just remembered that we did not have a conversation about this. Will you be plugging in the correct assets for this prior to merging to your internal repo?

Comment (EvanHahn-Signal) [reply SignalUI/Appearance/Theme+OWS.swift:301]: Yup, that's all sorted! Thanks for checking.

Comment (meherkasam) [reply SignalUI/Appearance/Theme+OWS.swift:301]: Thanks for the update. For future contributions, is leaving `TODO` comments in the code the best way to flag stuff like this or is there a more preferable way?

**Pull request closed without merging**